### PR TITLE
fix: Pick log message from merging object in Pino instrumentation

### LINKED ISSUE #2410: pino: logs in context decoration breaks mergingObject `msg` field when only mergingObject is passed in as an argument
## Description

When using the Node.js [agent decoration without forwarding option for Logs in Context](https://docs.newrelic.com/docs/logs/logs-context/configure-logs-context-nodejs/#2-decorate) with `pino`, if the [logger API](https://github.com/pinojs/pino/blob/main/docs/api.md#logger-instance) is called with a `mergingObject` instead of a string as the only argument, the library fails to take into account that the `msg` field in the object, always taking `args[1]` to [decorate](https://github.com/newrelic/node-newrelic/blob/main/lib/instrumentation/pino/pino.js#L59) and eventually setting `msg` as `undefined NR LINKING xxxx`. 

## Expected Behavior

Pino logs that are submitted as objects with a field `msg` or `message` have the `msg/message` field decorated with the NR LINKING attributes and set correctly. 

## Steps to Reproduce

Relevant section of `newrelic.cjs`:

```
exports.config = { 
//...
  application_logging: {
    forwarding: {
      enabled: false,
    },
    local_decorating: {
      enabled: true,
    },
  },
}
```

```
import Pino from 'pino';
const logger = Pino();
logger.error({msg: "mymsg", error: {attribute: "anattribute"}});
```

If you console log out `arguments` in the instrumentation [file](https://github.com/newrelic/node-newrelic/blob/main/lib/instrumentation/pino/pino.js#L51) when only a mergingObject is supplied, you get:

```
[Arguments] {
[node] '0': {
[node] msg: 'mymsg,
[node] error: {
[node] attribute: 'anattribute'
[node] }
[node] },
[node] '1': undefined, // <-- SEE HERE
[node] '2': 50,
[node] '3': ',"time":1722380481790'
[node] }
```

The resultant badly-decorated log looks like the following after the [metadata is added](https://github.com/newrelic/node-newrelic/blob/main/lib/instrumentation/pino/pino.js#L59).

```
{"level":50,"time":1722379746170,"pid":xxx,"hostname":"xxx","msg":"mymsg","error":{"attribute":"anattribute"},"msg":"undefined NR-LINKING|xxx|xxx|xxx|xxx|xxx|"}
```

This is opposed to just sending a string `logger.warn('a string'))`, which populates args[1] and generates a working log:

```
{
[node] '0': {},
[node] '1': 'a string',
[node] '2': 40,
[node] '3': ',"time":1722380481791'
[node] }
```
```
{"level":40,"time":1722380481791,"pid":xxx,"hostname":"xxx","msg":"a string NR-LINKING|xxx|xxx|xxx|xxx|xxx|"}
```

## Your Environment
newrelic 11.23.2
pino 9.3.2

## Additional Context

This is open with NewRelic support as Case #00215731

As a temporary workaround, we can send the `msg` portion as a second argument to pino in the form of `logger.error({ ...OUROBJECT }, msg)`, but that is cumbersome :) This behaviour is probably unexpected for a JSON logger that's looking for msg!

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR resolves #2410.

### CODE REVIEW

**Review by bizob2828 — CHANGES_REQUESTED** [review]
Overall looks good, just 2 small comments

File: test/versioned/pino/pino.tap.js [hunk 1]
```diff
@@ -83,14 +83,28 @@ tap.test('Pino instrumentation', (t) => {
...
         includeLocalDecorating: true,
         hostname: agent.config.getHostnameSafe(),
         logLine: line
       })
       t.same(agent.logs.getEvents(), [], 'should not add any logs to log aggregator')
+
+      // Verify that merging object only logs get decorated:
+      logger.info({ msg: message })
+      line = await once(stream, 'data')
+      console.log(line)
```
[the remark this answers is not in the document]
Comment (jsumners-nr) [reply test/versioned/pino/pino.tap.js:98]: Oops.

File: lib/instrumentation/pino/pino.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ const {
 } = require('../../util/application-logging')
 const semver = require('semver')
 
+// eslint-disable-next-line sonarjs/cognitive-complexity
```
[the remark this answers is not in the document]
Comment (jsumners-nr) [reply lib/instrumentation/pino/pino.js:18]: There are a lot of enclosed variables and the function itself uses the Pino instance for `this`. I think any refactor is going to be _more_ difficult to reason through. Also, what is this rule doing to determine complexity?

[the remark this answers is not in the document]
Comment (jsumners-nr) [reply lib/instrumentation/pino/pino.js:18]: In this particular case, the rule is saying the `instrument(shim, tools)` function has exceeded the threshold and that function has not been modified. The thing that was modified was an enclosed function: `wrappedAsJson`. The two should be considered separately, but it seems the tool is unable to distinguish between the two.

[the remark this answers is not in the document]
Comment (bizob2828) [reply lib/instrumentation/pino/pino.js:18]: It's based mostly on nesting functions. you can read more about it [here](https://medium.com/@himanshuganglani/clean-code-cognitive-complexity-by-sonarqube-659d49a6837d). we can defer but can you file a ticket to address?

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**